src/main/kotlin/store/storerepository/StoreRepositoryPostgreSql.kt

```diff
@@ -0,0 +1,138 @@
+package store.storerepository
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.jetbrains.exposed.sql.*
+import org.jetbrains.exposed.sql.ReferenceOption.CASCADE
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.joda.time.DateTime
+import store.domain.Store
+import store.domain.StoreInfo
+import store.domain.StoreRepository
+import java.sql.Blob
+import java.util.*
+import javax.sql.rowset.serial.SerialBlob
+
+// TODO should be dealing with real entities only
+class StoreRepositoryPostgreSql(private val database: Database) : StoreRepository {
+
+    init {
+        transaction(database) {
+            SchemaUtils.createMissingTablesAndColumns(
+                StoreSchema,
+                StoreExtraFieldSchema,
+            )
+        }
+    }
+
+    private object StoreSchema : Table("stores") {
+        val id = varchar("id", 36).primaryKey()
+        val externalId = varchar("external_id", 20).uniqueIndex()
+        val name = varchar("name", 50).nullable()
+        val customName = bool("custom_name").nullable()
+        val description = varchar("description", 2000).nullable()
+        val type = varchar("type", 40).nullable()
+        val openingDate = date("opening_date").nullable()
+        val code = varchar("code", 70).nullable()
+        val seasons = blob("seasons_json").nullable()
```

Comment:
supposing that we would need to have a filter on the frontend to display stores from a specific season - how could you implement this

Reply:
Since there were no requirements on the seasons, I decided to go Lean but I could:
- use PostgreSQL JSON abilities
- migrate the seasons to their own table
The chosen solution would depend on the expected data growth.
In both solutions, I'd have to pass the year and semester all along the way